### ISSUE TITLE
Incomplete search?

### ISSUE THREAD

**Issue by jdberry** [macports/macports-webapp#86]
When I search for the port "tag" in all categories (by entering the string "tag" at https://ports.macports.org), the "tag" port is not shown in the results. When I enter the same search-string after clicking in the sysutils category, then the tag port is displayed. Strange. I would expect the tag port to be found in the broader search.

**Comment by mojca** [comment]
What I suspect happens is that the search only displays the first N hits in some semi-random order. It doesn't explicitly attempt to check or display the exact hit first, so I assume it just gets (not) listed somewhere beyond the first 50(?) results.

Another problem I discovered with the search is that I can type a longer word, but the last letter I type is not always used or removed (somewhat depending on typing speed).

We should probably do two things. The first one is to special-case and show the exact match as the first hit. The second might be to potentially decrease the number of results on the first page (not sure) and the allow pagination to see the rest. Maybe the ports then need to be in alphabetical order, short of other more clever algorithms.

**Comment by jdberry** [comment]
Yeah, that's my guess too. Random order isn't very useful. Probably all results should be displayed, perhaps paged, in alphabetical order by port name. There might be an argument that any exact matches, or perhaps all prefix-matches, should be sorted out and displayed first, alphabetically. That would be a more complicated, though not too difficult, query.

**Comment by cjones051073** [comment]
Just ran into this so wondering if there had been any progress ? FWIW, I very much agree with the proposal to first list exact portname matches, and after that the partial matches, in a well defined (alphabetical) order. Currently, if you search for something that results in a lot of hits, finding what you want is quite awkward.

**Comment by mf2k** [comment]
I agree with the comments above that sorting the results alphabetically would help a lot with this.

**Comment by arjunsalyan** [comment]
I am working on a new search backend for the webapp. A demo is available here for feedback: http://macports.silentfox.tech/
